### DISCUSSION TITLE
Global helper function for environment variables

### DISCUSSION THREAD

**Discussion by jasonmccallister** [craftcms/cms#8377] [Ideas]
### Description

In Laravel, you can access the environment variables with the `env('key')` function. The most convenient use for this function is that you can also pass a default value as a second argument, so if the environment key is null, the default would be used. 

This would be great for multi-environment configs as you can specify the defaults (think of the defaults as always setting `devMode` to false unless specifically defined in the environment). 

For example, the `general.php` would look like this:

```php
<?php

return [
    'devMode' => env('APP_DEV_MODE', false)
];
```

The function definition in Laravel exists [here](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Foundation/helpers.php#L429) but looks like this:

```php
if (! function_exists('env')) {
    /**
     * Gets the value of an environment variable.
     *
     * @param  string  $key
     * @param  mixed   $default
     * @return mixed
     */
    function env($key, $default = null)
    {
        $value = getenv($key);
        if ($value === false) {
            return value($default);
        }
        switch (strtolower($value)) {
            case 'true':
            case '(true)':
                return true;
            case 'false':
            case '(false)':
                return false;
            case 'empty':
            case '(empty)':
                return '';
            case 'null':
            case '(null)':
                return;
        }
        if (strlen($value) > 1 && Str::startsWith($value, '"') && Str::endsWith($value, '"')) {
            return substr($value, 1, -1);
        }
        return $value;
    }
}
```

This would have a lot of benefits, but one specific would be not needing to specify environment variables for production as the "sensible defaults" would be, well, default.

**Comment by dmlogic** [comment]
+1

I'm already doing precisely the above with my Craft 2 apps.
